src/http-client.js

```diff
@@ -43,6 +43,11 @@ export class HttpClient {
     if (typeof fetch === 'undefined') {
       throw new Error('HttpClient requires a Fetch API implementation, but the current environment doesn\'t support it. You may need to load a polyfill such as https://github.com/github/fetch');
     }
+
+    // Use application/json as the default content-type.
+    this.defaults = {
+      headers: { 'content-type': 'application/json' }
```

Comment:
@davismj This does not play well with CORS. Some API endpoints do not accept the `content-type` header on requests. Got a CORS error after upgrading to the version that includes this.

Reply:
Hm. I'm thinking that is the right thing to do anyway, and that the problem isn't the library but your CORS configuration. However, I think you're right that maybe we need to only add the header when the post has a body. 

In any case, if you're posting with a body to a cross domain API, you'd really want to respond with content-type in your Access-Control-Allow-Headers list, and if it was working before, it was more of a coincidence.

Don't you think?